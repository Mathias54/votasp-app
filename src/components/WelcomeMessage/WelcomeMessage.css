.welcome-text {
  color: #333;
  text-align: center;
}
.welcome-text h2{
  font-size: 30px;
}

.welcome-text p {
  font-size: 16px;
  line-height: 25px;
  margin-bottom: 50px;
}
