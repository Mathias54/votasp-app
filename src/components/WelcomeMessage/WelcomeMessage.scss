.welcome-text {
	color: #333;
  text-align: center;

  h2 {
    font-size: 30px;
  }

  p {
    line-height: 25px;
    font-size: 16px;
    margin-bottom: 50px;
  }
}
