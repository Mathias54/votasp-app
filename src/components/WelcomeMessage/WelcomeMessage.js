import React from 'react';
import { Link } from 'react-router-dom';
import './WelcomeMessage.css';

const regions = {
  sp: { name: 'São Paulo', projectName: 'VotaSP' }
};

const WelcomeMessage = ({ regionTag }) => {
  const region = regions[regionTag] || regions['sp'];

  return (
    <section>
      <div className="container welcome-text">
        <h2>
          <strong>BEM-VINDO</strong>
        </h2>
        <p>
          O projeto {region.projectName} nasceu com o objetivo de ajudar a
          população do estado de {region.name} a escolher seus candidatos.
          Através de questões objetivas os eleitores podem verificar suas
          afinidades com os candidatos.<br />
          <strong>É muito simples e prático!</strong>
        </p>

        <Link to={`/perguntas`} className="btn btn-primary">
          Começar
        </Link>
      </div>
    </section>
  );
};

export default WelcomeMessage;
