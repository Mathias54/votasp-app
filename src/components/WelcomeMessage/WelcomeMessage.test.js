import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WelcomeMessage from './WelcomeMessage';

describe('<WelcomeMessage />', () => {
  it('should match snapshot', () => {
    const welcomeMessage = shallow((
      <WelcomeMessage />
    ))

    expect(toJson(welcomeMessage)).toMatchSnapshot();
  });

  it('should accept valid regionTag', () => {
    const welcomeMessage = shallow((
      <WelcomeMessage regionTag='sp' />
    ))

    expect(toJson(welcomeMessage)).toMatchSnapshot();
  });
});
