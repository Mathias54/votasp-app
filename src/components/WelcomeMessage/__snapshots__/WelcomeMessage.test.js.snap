// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeMessage /> should match snapshot 1`] = `
<section>
  <div
    className="container welcome-text"
  >
    <h2>
      <strong>
        BEM-VINDO
      </strong>
    </h2>
    <p>
      O projeto 
      VotaSP
       nasceu com o objetivo de ajudar a população do estado de 
      São Paulo
       a escolher seus candidatos. Através de questões objetivas os eleitores podem verificar suas afinidades com os candidatos.
      <br />
      <strong>
        É muito simples e prático!
      </strong>
    </p>
    <Link
      className="btn btn-primary"
      replace={false}
      to="/perguntas"
    >
      Começar
    </Link>
  </div>
</section>
`;
